custom-war-packager-maven-plugin/src/main/java/io/jenkins/tools/warpackager/mavenplugin/BuildMojo.java

```diff
@@ -0,0 +1,105 @@
+package io.jenkins.tools.warpackager.mavenplugin;
+
+import io.jenkins.tools.warpackager.lib.config.Config;
+import io.jenkins.tools.warpackager.lib.impl.Builder;
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+import java.io.File;
+import java.io.IOException;
+
+import static org.apache.maven.plugins.annotations.LifecyclePhase.PACKAGE;
+import static org.apache.maven.plugins.annotations.ResolutionScope.RUNTIME;
+
+//TODO: There is a serious correlation with Maven HPI's Plugin "custom-war" step. This step is actually used inside via Maven-in-Maven. May be reworked later.
+
+/**
+ * Mojo wrapper for {@link Builder}.
+ * @author Oleg Nenashev
+ * @since TODO
+ */
+@Mojo(name="build", defaultPhase = PACKAGE, requiresProject = false)
+public class BuildMojo extends AbstractMojo {
+
+    @Parameter(property = "configFile", required = true)
+    public @CheckForNull String configFilePath;
+
+    @Parameter(property = "version")
+    public @CheckForNull String warVersion;
+
+    @Parameter(property = "tmpDir")
```

Comment:
BTW if you add Javadoc for these things, and configure your POM to generate `HelpMojo`, you will make it easier for people to invoke this.

Reply:
For sure. Created https://issues.jenkins-ci.org/browse/JENKINS-51259